front/src/components/LoginPanel/comp/RegisterForm.test.js

```diff
@@ -0,0 +1,35 @@
+import { render, screen } from "@testing-library/react";
+import RegisterForm from "./RegisterForm";
+import { ApolloProvider } from "@apollo/client/react";
+import { ApolloClient, InMemoryCache } from "@apollo/client";
+
+const client = new ApolloClient({
+    uri: "http://localhost:3001/graphql",
+    cache: new InMemoryCache(),
+});
+
+describe("RegisterForm", () => {
+    beforeEach(() => {
+        render(<ApolloProvider client={client}>
+            <RegisterForm />
+        </ApolloProvider>
+        );
+    });
+
+    describe("should be one register form", () => {
+        it("should have three text inputs", () => {
+            let textInputs = screen.getAllByRole("textbox");
+            expect(textInputs).toHaveLength(3);
+        });
+    
+        it("should have one textbox", () => {
+            let textbox = screen.getByPlaceholderText("Dlaczego chcesz założyć konto?");
+            expect(textbox).toBeInTheDocument();
+        });
+    
+        it("should have one button", () => {
+            let buttons = screen.getAllByRole("button");
+            expect(buttons).toHaveLength(1);
+        });

```

Comment:
It is very easy to break something and as long as there is a single button rendered, this test will pass. Tests should check some specific, interesting or important property of the component, if it correctly changes states etc.
Here we can simply ask for a button with given name:
```suggestion
it("should have register button", () => {
            let btn = screen.getByRole("button", { name: "Zarejestruj się" });
            expect(btn).toBeInTheDocument();
        });
```
Note use of 'getByRole' instead of 'getAllByRole' - it asserts that only one result is found and makes expect(buttons).toHaveLength(1); redundant.